Mathlib/Data/Bitvec/Lemmas.lean

```diff
@@ -124,4 +130,71 @@ theorem ofFin_toFin {n} (v : BitVec n) : ofFin (toFin v) = v := by
   rfl
 #align bitvec.of_fin_to_fin Std.BitVec.ofFin_toFin
 
+/-!
+### Distributivity of `Std.BitVec.ofFin`
+-/
+section
+variable (x y : Fin (2^w))
+
+@[simp] lemma ofFin_neg : ofFin (-x) = -(ofFin x) := by
+  rw [neg_eq_zero_sub]; rfl
+
+@[simp] lemma ofFin_and : ofFin (x &&& y) = ofFin x &&& ofFin y := rfl
+@[simp] lemma ofFin_or  : ofFin (x ||| y) = ofFin x ||| ofFin y := rfl
+@[simp] lemma ofFin_xor : ofFin (x ^^^ y) = ofFin x ^^^ ofFin y := rfl
+@[simp] lemma ofFin_add : ofFin (x + y)   = ofFin x + ofFin y   := rfl
+@[simp] lemma ofFin_sub : ofFin (x - y)   = ofFin x - ofFin y   := rfl
+@[simp] lemma ofFin_mul : ofFin (x * y)   = ofFin x * ofFin y   := rfl
+
+@[simp] lemma ofFin_zero    : @ofFin w 0 = 0 := rfl
+@[simp] lemma ofFin_one     : @ofFin w 1 = 1 := rfl
+
+end
+
+/-!
+### Distributivity of `Std.BitVec.toFin`
+-/
+section
+variable (x y : BitVec w)
+
+@[simp] lemma toFin_neg : toFin (-x) = -(toFin x) := by
+  rw [neg_eq_zero_sub]; rfl
+
+@[simp] lemma toFin_and : toFin (x &&& y) = toFin x &&& toFin y := rfl
+@[simp] lemma toFin_or  : toFin (x ||| y) = toFin x ||| toFin y := rfl
+@[simp] lemma toFin_xor : toFin (x ^^^ y) = toFin x ^^^ toFin y := rfl
+@[simp] lemma toFin_add : toFin (x + y)   = toFin x + toFin y   := rfl
+@[simp] lemma toFin_sub : toFin (x - y)   = toFin x - toFin y   := rfl
+@[simp] lemma toFin_mul : toFin (x * y)   = toFin x * toFin y   := rfl
+
+@[simp] lemma toFin_zero    : @toFin w 0 = 0 := rfl
+@[simp] lemma toFin_one     : @toFin w 1 = 1 := rfl
```

Comment:
This looks like a bad choice of simp lemma by std, so I'd be inclined to leave a TODO comment (after removing `simp`)